feat: new sharebuttons

maehr opened the pull request:
# Pull request

## Proposed changes

- Fix #98 
- Replace dependency
- Better styling

<!-- Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. -->

## Types of changes

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] New feature (non-breaking change which adds functionality).
- [x] Enhancement (non-breaking change which enhances functionality)
- [ ] Bug Fix (non-breaking change which fixes an issue).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have read the **[README](./README.md)** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a unified `ShareButtons` component for easy content sharing across email, Facebook, Twitter, WhatsApp, and LinkedIn. Utilizes the Web Share API for enhanced sharing capabilities on supported browsers.
- **Refactor**
	- Updated the blog post layout for improved content presentation.
	- Simplified social media sharing by replacing specific share button components with a generic `ShareButtons` component across all blog posts.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

koilebeit reviewed (commented):
- On Firefox I only see one button where you can't read the font either:

![image](https://github.com/Stadt-Geschichte-Basel/stadtgeschichtebasel.ch/assets/73133626/b2235a6a-6ec2-4b8f-a1e5-848239e06113)

- On Brave there are all buttons, but on hover the buttons almost disappear

maehr commented:
> * On Firefox I only see one button where you can't read the font either:
> 
> ![image](https://private-user-images.githubusercontent.com/73133626/309537076-b2235a6a-6ec2-4b8f-a1e5-848239e06113.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk0NTY4MTgsIm5iZiI6MTcwOTQ1NjUxOCwicGF0aCI6Ii83MzEzMzYyNi8zMDk1MzcwNzYtYjIyMzVhNmEtNmVjMi00YjhmLWExZTUtODQ4MjM5ZTA2MTEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAzVDA5MDE1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlZWNhMmI4Mzc1OWI3MTVlYTMwMDA5ZjYyYTU0NzY1YjA2YmEwZTU1MzM4ZGE0MDhmMmJiYjM2NTgxMzRiYzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.M2PEQzVJ3VuBiOrw4vz3cPI7Y5yrL62t19-v7nVcoGM)
> 
> * On Brave there are all buttons, but on hover the buttons almost disappear

I did not check for Firefox, on Chrome I did experience the same issue with the buttons. I am not yet sure what causes it...

mtwente commented:
I changed the button styles so they do not disappear (fill up with white) on hover. I can see the buttons in Chrome and Firefox just fine.

Additionally, I made them smaller and changed the colour of the logos to secondary (yellow) on hover.

<img width="708" alt="smaller buttons with hover-colouring" src="https://github.com/Stadt-Geschichte-Basel/stadtgeschichtebasel.ch/assets/127845092/70b41354-b9fb-41ab-9ee7-182f4f0107be">

maehr commented:
> I changed the button styles so they do not disappear (fill up with white) on hover. I can see the buttons in Chrome and Firefox just fine.
> 
> Additionally, I made them smaller and changed the colour of the logos to secondary (yellow) on hover.
> 
> <img alt="smaller buttons with hover-colouring" width="708" src="https://private-user-images.githubusercontent.com/127845092/309541290-70b41354-b9fb-41ab-9ee7-182f4f0107be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk0NjEwNTQsIm5iZiI6MTcwOTQ2MDc1NCwicGF0aCI6Ii8xMjc4NDUwOTIvMzA5NTQxMjkwLTcwYjQxMzU0LWI5ZmItNDFhYi05ZWU3LTE4MmY0ZjAxMDdiZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwM1QxMDEyMzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NTE0MGZiZmJhZDZhODNmYzhjYmFhMWQyYmY4NTI1ZTkwN2Y3MmU2YTEwZDFiYmU2YjE4YjQ5NDBiYWZkZDc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.5recDgKMNJt7fdZzfqxjoWJFi0X-pJsjfidXNRsc0hM">

I cannot see your changes. (You need to use this https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request)
I pushed some of my own.

mtwente commented on src/lib/components/ShareButtons.svelte:
```diff
@@ -0,0 +1,149 @@
+			>
+		</a>
+		<a
+			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:scale-125"
```
```suggestion
			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:fill-secondary-500"
```

mtwente commented on src/lib/components/ShareButtons.svelte:
```diff
@@ -0,0 +1,149 @@
+			>
+		</a>
+		<a
+			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:scale-125"
```
```suggestion
			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:fill-secondary-500"
```

mtwente commented on src/lib/components/ShareButtons.svelte:
```diff
@@ -0,0 +1,149 @@
+			>
+		</a>
+		<a
+			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:scale-125"
```
```suggestion
			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:fill-secondary-500"
```

mtwente commented on src/lib/components/ShareButtons.svelte:
```diff
@@ -0,0 +1,149 @@
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 512 512"
+					aria-hidden="true"
+					class="h-6 w-6"
```
```suggestion
					class="h-4 w-4"
```

mtwente commented on src/lib/components/ShareButtons.svelte:
```diff
@@ -0,0 +1,149 @@
+					aria-hidden="true"
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 320 512"
+					class="h-6 w-6"
```
```suggestion
					class="h-4 w-4"
```

mtwente commented on src/lib/components/ShareButtons.svelte:
```diff
@@ -0,0 +1,149 @@
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 512 512"
+					aria-hidden="true"
+					class="h-6 w-6"
```
```suggestion
					class="h-4 w-4"
```

mtwente commented on src/lib/components/ShareButtons.svelte:
```diff
@@ -0,0 +1,149 @@
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 448 512"
+					aria-hidden="true"
+					class="h-6 w-6"
```
```suggestion
					class="h-4 w-4"
```

mtwente commented on src/lib/components/ShareButtons.svelte:
```diff
@@ -0,0 +1,149 @@
+					aria-hidden="true"
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 448 512"
+					class="h-6 w-6"
```
```suggestion
					class="h-4 w-4"
```

mtwente commented on src/lib/components/ShareButtons.svelte:
```diff
@@ -0,0 +1,149 @@
+		class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:scale-125"
+		on:click={handleWebShare}
+	>
+		<svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" viewBox="0 0 512 512"
```
```suggestion
		<svg aria-hidden="true"
		xmlns="http://www.w3.org/2000/svg"
		class="h-4 w-4"
		viewBox="0 0 512 512"
```

mtwente commented on src/routes/blog/[slug]/+page.svelte:
```diff
@@ -28,22 +27,18 @@
-				<a
-					class="variant-ringed badge mr-1 text-primary-500 hover:bg-primary-500 hover:text-white"
-					href="{base}/blog/kategorie/{category}">&num;{category}</a
+			<div class="flex flex-col flex-wrap gap-1 space-y-2 md:flex-row md:place-items-center">
```
```suggestion
			<div class="flex flex-col flex-wrap items-start gap-1 space-y-1 md:flex-row md:place-items-center">
```

mtwente commented on src/lib/components/ShareButtons.svelte:
```diff
@@ -0,0 +1,149 @@
+	</div>
+{:else}
+	<button
+		class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:scale-125"
```
```suggestion
		class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:fill-secondary-500"
```

mtwente reviewed (changes requested):
Apart from styling changes, I made a suggestion to prevent this behaviour in  `src/routes/blog/[slug]/+page.svelte`.

Did we say no to a Telegram button?

![localhost_5173_blog_hag-laedt-zu-vortraegen-am-4-maerz-2024-ein (1)](https://github.com/Stadt-Geschichte-Basel/stadtgeschichtebasel.ch/assets/127845092/e4c0a1f2-dce6-4f50-8d79-19c7da6bc289)

maehr commented:
> Apart from styling changes, I made a suggestion to prevent this behaviour in `src/routes/blog/[slug]/+page.svelte`.
> 
> Did we say no to a Telegram button?
> 
> ![localhost_5173_blog_hag-laedt-zu-vortraegen-am-4-maerz-2024-ein (1)](https://private-user-images.githubusercontent.com/127845092/309558031-e4c0a1f2-dce6-4f50-8d79-19c7da6bc289.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk0ODE0NjEsIm5iZiI6MTcwOTQ4MTE2MSwicGF0aCI6Ii8xMjc4NDUwOTIvMzA5NTU4MDMxLWU0YzBhMWYyLWRjZTYtNGY1MC04ZDc5LTE5YzdkYTZiYzI4OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwM1QxNTUyNDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NWYzODc1ZWIwMTBiYTA3ZDY3NzUyMGViMWY4ZTA0ZDczOTA2ZGE3NTA2MTA3OWIyYjcyNzM5YzJlMjAzZWQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.DE4l--GDBCxaSS4VO-AC8yNNDQdQ2ZAO7LK6tJbLWgc)

Thanks for the review and the suggestions.
PS: Yes, we won't be implementing Telegram.
